ludwig/benchmarking/utils.py

```diff
@@ -0,0 +1,125 @@
+import copy
+import logging
+import os
+import shutil
+from datetime import datetime
+from types import ModuleType
+from typing import Any, Dict, Tuple, Union
+
+import fsspec
+import pandas as pd
+from botocore.exceptions import ClientError
+from globals import CONFIG_YAML, EXPERIMENT_RUN, REPORT_JSON
+from s3fs.errors import translate_boto_error
+
+from ludwig.constants import CATEGORY
+from ludwig.globals import MODEL_HYPERPARAMETERS_FILE_NAME
+from ludwig.utils.dataset_utils import get_repeatable_train_val_test_split
+from ludwig.utils.defaults import default_random_seed, merge_with_defaults
+from ludwig.utils.fs_utils import get_fs_and_path
+
+
+def get_full_config(config: Dict[str, Any]) -> Dict[str, Any]:
+    """Return a full Ludwig config given an experiment config.
+
+    config: ludwig config.
+    """
+    config_dict = copy.deepcopy(config)
+    config = merge_with_defaults(config_dict)
+    return config
+
+
+def load_from_module(
+    dataset_module: ModuleType, output_feature: Dict[str, str], subsample_frac: float = 1
+) -> Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame, pd.DataFrame]:
+    """Take in Ludwig Python module of the dataset, load it, and split it if it isn't already (by adding "split"
+    column).
+
+    dataset_module: ludwig dataset module (e.g. ludwig.datasets.sst2, ludwig.datasets.ames_housing, etc.)
+    subsample_frac: percentage of the total datasets to load.
+    """
+    dataset = dataset_module.load(split=False)
+    if subsample_frac < 1:
+        dataset = dataset.sample(frac=subsample_frac, replace=False, random_state=default_random_seed)
+
+    if output_feature["type"] == CATEGORY:
+        dataset = get_repeatable_train_val_test_split(
+            dataset,
+            stratify_colname=output_feature["name"],
+            random_seed=default_random_seed,
+            frac_train=0.75,
+            frac_val=0.1,
+            frac_test=0.15,
```

Comment:
Is there a strong reason to not go with the defaults of 0.7, 0.2, and 0.1 that's the ludwig and get_repeatable_train_val_test_split default?

Reply:
No, will switch to defaults.